avm/res/azure-stack-hci/cluster/tests/e2e/2nodeswitched.defaults/main.test.bicep

```diff
@@ -0,0 +1,304 @@
+targetScope = 'subscription'
+
+metadata name = 'Deploy Azure Stack HCI Cluster in Azure with a 2 node switched configuration'
+metadata description = 'This test deploys an Azure VM to host a 2 node switched Azure Stack HCI cluster, validates the cluster configuration, and then deploys the cluster.'
+
+@description('Optional. The name of the Azure Stack HCI cluster - this must be a valid Active Directory computer name and will be the name of your cluster in Azure.')
+@maxLength(15)
+@minLength(4)
+param name string = 'hcicluster'
+@description('Optional. Location for all resources.')
+param location string = deployment().location
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-azure-stack-hci.cluster-${serviceShort}-rg'
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'ashc2nmin'
+@description('Optional. A token to inject into the name of each resource.')
+param namePrefix string = '#_namePrefix_#'
+@minLength(4)
+@maxLength(8)
+@description('Optional. The prefix for the resource for the deployment. This value is used in key vault and storage account names in this template, as well as for the deploymentSettings.properties.deploymentConfiguration.scaleUnits.deploymentData.namingPrefix property which requires regex pattern: ^[a-zA-Z0-9-]{1,8}$.')
+param deploymentPrefix string = take('${take(namePrefix, 8)}${uniqueString(utcNow())}', 8)
+@description('Optional. The username of the LCM deployment user created in Active Directory.')
+param deploymentUsername string = 'deployUser'
```

Comment:
We should only have extra parameters if absolutely necessary. Please move all that are not needed / could be moved into the `dependencies.bicep` files / could be referenced directly in the testDeployment / be in a variable to these places.

Reply:
I strongly encourage using other module's test cases as a reference. Also, the same comment applies to the other test cases.